packages/main/test/specs/Select.spec.js

```diff
@@ -188,6 +188,33 @@ describe("Select general interaction", () => {
 		assert.strictEqual(await inputResult.getProperty("value"), "3", "Change event should have fired twice");
 	});
 
+	it("remains closed and unchanged when read-only", async () => {
+		const select = await browser.$("#mySelectReadOnly");
+		const EXPECTED_SELECTION_TEXT = "Compact";
+		const selectOptionText = await select.shadow$(".ui5-select-label-root");
+
+		// act - try to open the dropdown
+		await select.click();
+
+		const staticAreaItemClassName = await browser.getStaticAreaItemClassName("#mySelectReadOnly");
+		const popover = await browser.$(`.${staticAreaItemClassName}`).shadow$("ui5-responsive-popover");
+
+		// assert
+		assert.notOk(await popover.getProperty("opened"), "Select remains closed.");
```

Comment:
Can we also coverage if the selectedOption is the same using selectedOption getter. Also a test case for SelectMenu.